packages/angular_devkit/build_angular/src/babel/presets/application.ts

```diff
@@ -124,21 +127,25 @@ export default function (api: unknown, options: ApplicationPresetOptions) {
   const plugins = [];
   let needRuntimeTransform = false;
 
-  if (options.angularLinker) {
+  if (options.angularLinker?.shouldLink) {
     // Babel currently is synchronous so import cannot be used
     const {
       createEs2015LinkerPlugin,
-    } = require('@angular/compiler-cli/linker/babel');
+    } = require('@angular/compiler-cli/linker/babel') as typeof import('@angular/compiler-cli/linker/babel');
 
     plugins.push(createEs2015LinkerPlugin({
+      linkerJitMode: options.angularLinker?.jitMode,
```

Comment:
```suggestion
      linkerJitMode: options.angularLinker.jitMode,
```
nit: null check shouldn't be need since `angularLinker` needs to be defined to enter the if block